core/trino-main/src/main/java/io/trino/sql/analyzer/FeaturesConfig.java

```diff
@@ -141,6 +141,8 @@
     private int filterAndProjectMinOutputPageRowCount = 256;
     private int maxGroupingSets = 2048;
 
+    private boolean legacyCatalogRoles;
```

Comment:
I think legacy mode should be default. So far there are no system access control that handle roles (other than file based), so I think we should cause any issues with updates with this.

If you agree then let's rename toggle to `experimental.system-roles` or something like this.

Reply:
> I think legacy mode should be default.

No, legacy mode should be legacy, and as such, it should be disabled by default. If anyone is relying on the old behavior, they can fall back to the legacy version by setting the toggle. Eventually, we want to remove the legacy form altogether.